kernelci/config/__init__.py

```diff
@@ -78,6 +78,46 @@ def load_yaml(config_path, validate_entries=None):
     return config
 
 
+def _merge_trees(a, b):
+    """Merge two values loaded from YAML"""
+    if isinstance(a, dict) and isinstance(b, dict):
+        all_keys = set(a.keys())
+        all_keys.update(b.keys())
+        all_keys = list(all_keys)
+        res = dict()
+        for k in all_keys:
```

Comment:
Maybe we could use set operators here?

```suggestion
        res = dict()
        for k in (set(a) | set(b)):
```